types/chain_info.go

```diff
@@ -0,0 +1,35 @@
+package types
+
+import (
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+// ChainInfo is used to track metadata about a peer and its chain.
+type ChainInfo struct {
+	Peer   peer.ID
+	Head   TipSetKey
+	Height uint64
```

Comment:
Mentioned in slack, a `Trusted bool` filed on this would ease conflicts later (#3167), can just hardcode it to `true` as nothing uses yet.